### PR TITLE
New Parser for show ip nbar protocol discovery

### PR DESCRIPTION
**Pull request by gwoodwa1**
Unitests run successfully without any errors:

(sdk) gary@gary-MS-7B17:~/sdk/genieparser/src/genie/libs/parser/iosxe/tests$ python -m unittest discover
.................................................
----------------------------------------------------------------------
Ran 49 tests in 0.485s

OK

### CODE REVIEW

File: src/genie/libs/parser/iosxe/tests/ShowIpNbarDiscovery/cli/equal/golden_output1_output.txt [hunk 1]
```diff
@@ -0,0 +1,77 @@
+unknown                      3544                     15
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/tests/ShowIpNbarDiscovery/cli/equal/golden_output1_output.txt:1]: it looks like output is snipped. missing `GigabitEthernet1` and other lines.

File: src/genie/libs/parser/iosxe/show_nbar.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    schema = {
+        'interface': {
+            Any(): {
+                'protocol': {
+                    Any(): {
+                        'IN Packet Count': int,
+                        'OUT Packet Count': int,
+                        'IN Byte Count': int,
+                        'OUT Byte Count': int,
+                        'IN 5min Bit Rate (bps)': int,
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/show_nbar.py:20]: key name is lower case without space and brackets. so, please update to like `in_5min_bit_rate_bps`. others as well where applicable.

File: src/genie/libs/parser/iosxe/show_nbar.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        p1 = re.compile(r'^(?P<interface>Gig.+|Ten.+|Fast.+ |Port.+)')
+        p2 = re.compile(r'^(?P<protocol>[\w\-]+) +(?P<In_Packet_Count>[\d]+) +(?P<Out_Packet_Count>[\d]+)')
+        p3 = re.compile(r'^(?P<In_Byte_Count>[\d]+) +(?P<Out_Byte_Count>[\d]+)')
+        p4 = re.compile(r'^(?P<In_Bitrate>[\d]+) +(?P<Out_Bitrate>[\d]+)')
+        p5 = re.compile(r'^(?P<In_Bitrate_Max>[\d]+) +(?P<Out_Bitrate_Max>[\d]+)')
+
+        
+
+        for line in out.splitlines():
+            #import pdb; pdb.set_trace()
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/show_nbar.py:64]: remove if not needed anymore. line 67 as well.

**Review by tahigash — CHANGES_REQUESTED** [review]
thank you for your contribution! reviewed schema mainly.

File: src/genie/libs/parser/iosxe/show_nbar.py (hunk 3 above)
Comment (gwoodwa1) [reply src/genie/libs/parser/iosxe/show_nbar.py:64]: I'll remove the left over pdb lines. Good spot.

File: src/genie/libs/parser/iosxe/tests/ShowIpNbarDiscovery/cli/equal/golden_output1_output.txt (hunk 1 above)
Comment (gwoodwa1) [reply src/genie/libs/parser/iosxe/tests/ShowIpNbarDiscovery/cli/equal/golden_output1_output.txt:1]: Yes looks like a copy/paste issue. I'll fix that one.

**Comment by gwoodwa1** [comment]
All amended files uploaded - also re-ran the unittest (below)
(sdk) gary@gary-MS-7B17:~/sdk/genieparser/tests/iosxe$ python -m unittest discover
.................................................
----------------------------------------------------------------------
Ran 49 tests in 0.463s

OK

File: src/genie/libs/parser/iosxe/show_nbar.py (hunk 2 above)
Comment (gwoodwa1) [reply src/genie/libs/parser/iosxe/show_nbar.py]: Fixed, all changed to lower case. Thank you.

File: src/genie/libs/parser/iosxe/tests/ShowIpNbarDiscovery/cli/equal/golden_output1_output.txt (hunk 1 above)
Comment (tahigash) [reply src/genie/libs/parser/iosxe/tests/ShowIpNbarDiscovery/cli/equal/golden_output1_output.txt]: confirmed the fix

File: src/genie/libs/parser/iosxe/show_nbar.py (hunk 2 above)
Comment (tahigash) [reply src/genie/libs/parser/iosxe/show_nbar.py]: confirmed the change

File: src/genie/libs/parser/iosxe/show_nbar.py (hunk 3 above)
Comment (tahigash) [reply src/genie/libs/parser/iosxe/show_nbar.py]: confirmed the change

**Review by tahigash — APPROVED** [review]

**Review by choudnik — APPROVED** [review]

**Pull request merged**